sandy/core/cov.py

```diff
@@ -812,22 +941,35 @@ def _constrained_gls_sensitivity(self, S, threshold=None):
                       0	              1
         0	-2.00000e+00	1.50000e+00
         1	1.00000e+00	  -5.00000e-01
+        >>> sensitivity._constrained_gls_sensitivity(S, sparse=True)
+                      0	              1
+        0	-2.00000e+00	1.50000e+00
+        1	1.00000e+00	  -5.00000e-01
         """
         # Data in a appropiate format
-        Vx_prior = self.data.values
-        Vy_extra = pd.DataFrame(0, index=self.data.index,
-                                columns=self.data.columns)
         S_ = pd.DataFrame(S)
         index = S_.index
         columns = S_.columns
-        # constrained Least Squares sensitivity
-        G_inv = self._gls_G_inv(S, Vy_extra).values
-        sensitivity = G_inv.dot(S_.values).dot(Vx_prior)
+        if sparse is True:
+            Vx_prior = self.to_sparse()
+            S_ = sps.csr_matrix(S_.values)
+            G = self._gls_Vy_calc(S, sparse=sparse)
+            G_inv = sandy.CategoryCov(G).invert(sparse=sparse)
+            G_inv =  sps.csr_matrix(G_inv.data.values)
+            # constrained Least Squares sensitivity
+            sensitivity = G_inv.dot(S_).dot(Vx_prior).toarray()
+        else:
+            Vx_prior = self.data.values
+            Vy_extra = pd.DataFrame(0, index=self.data.index,
+                                    columns=self.data.columns)
+            # constrained Least Squares sensitivity

```

Comment:
Instead of passing a df with zeros, should Vy_extra be optional in _gls_G_inv?

Reply:
Better idea. So i will correct the above comment also